sklearn/impute/_base.py

```diff
@@ -228,7 +228,11 @@ def _validate_input(self, X, in_fit):
                                                         self.strategy))
 
         if self.strategy in ("most_frequent", "constant"):
-            dtype = None
+            if isinstance(X, (list, tuple)) and \
+               any(isinstance(elem, str) for row in X for elem in row):
+                dtype = np.object
```

Comment:
For this particular case, I prefer `object` instead of `np.object` (similarly to testing functions). Moreover, adding a comment why is this data type required would be helpful.